packages/x-date-pickers/src/DateField/tests/selection.DateField.test.tsx

```diff
@@ -77,7 +77,7 @@ describe('<DateField /> - Selection', () => {
 
     it('should select day on mobile', () => {
       render(<DateField />);
-      const input: HTMLInputElement = screen.getByRole('textbox');
+      const input: HTMLInputElement = screen.getByRole<HTMLInputElement>('textbox');
```

Comment:
If we are going with an explicit way, probably double typing is already redundant. 🙈 
```suggestion
      const input = screen.getByRole<HTMLInputElement>('textbox');
```